DIP/DIP/Models/CardPaymentService.cs

```diff
@@ -5,14 +5,19 @@ namespace DIP
 {
     public class CardPaymentService : IPaymentService<CardPayment>
     {
+        private LogBase _logBase = new FileLogger();
         public string Pay(CardPayment payment )
         {
             if (ValidDiscount(payment.Discount)) {
```

Comment:
Rename the method to validate discount, let it throw the exception since this is not the responsibility of the Pay method. and remove the if else block